File: frontend/admin/src/core/api-client.ts
import axios from 'axios';
import type { AxiosInstance, AxiosRequestConfig } from 'axios';
import type { ApiResponse } from './api-response';
import { API_BASE_URL } from './constants';

export class ApiClient {
  private readonly client: AxiosInstance;

  constructor(baseURL: string) {
    this.client = axios.create({
      baseURL,
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    });

    this.client.interceptors.request.use((config) => {
      config.headers.set('X-Request-Id', this.generateRequestId());
      return config;
    });
  }

  async get<T>(path: string, config?: AxiosRequestConfig): Promise<ApiResponse<T>> {
    const response = await this.client.get<ApiResponse<T>>(path, config);
    return response.data;
  }

  async post<T>(
    path: string,
    data?: unknown,
    config?: AxiosRequestConfig,
  ): Promise<ApiResponse<T>> {
    const response = await this.client.post<ApiResponse<T>>(path, data, config);
    return response.data;
  }

  async put<T>(
    path: string,
    data?: unknown,
    config?: AxiosRequestConfig,
  ): Promise<ApiResponse<T>> {
    const response = await this.client.put<ApiResponse<T>>(path, data, config);
    return response.data;
  }

  async delete<T>(path: string, config?: AxiosRequestConfig): Promise<ApiResponse<T>> {
    const response = await this.client.delete<ApiResponse<T>>(path, config);
    return response.data;
  }

  private generateRequestId(): string {
    return `BEZA-${Date.now()}-${Math.random().toString(36).substring(2, 7)}`;
  }
}

export const apiClient = new ApiClient(API_BASE_URL);

File: frontend/mobile/lib/core/api_client.dart
import 'dart:convert';
import 'dart:io';
import 'api_response.dart';

class ApiClient {
  final String baseUrl;
  final Map<String, String> defaultHeaders;
  final HttpClient Function()? httpClientFactory;

  ApiClient({
    required this.baseUrl,
    this.defaultHeaders = const {},
    this.httpClientFactory,
  });

  Future<ApiResponse> get(String path, {Map<String, String>? headers}) async {
    final client = httpClientFactory?.call() ?? HttpClient();
    try {
      final request = await client.getUrl(Uri.parse('$baseUrl$path'));
      _addHeaders(request, headers);
      final response = await request.close();
      final body = await response.transform(utf8.decoder).join();
      return ApiResponse.fromJson(
        json.decode(body) as Map<String, dynamic>,
        null,
      );
    } finally {
      client.close();
    }
  }

  Future<ApiResponse> post(
    String path, {
    Map<String, dynamic>? body,
    Map<String, String>? headers,
  }) async {
    final client = httpClientFactory?.call() ?? HttpClient();
    try {
      final request = await client.postUrl(Uri.parse('$baseUrl$path'));
      _addHeaders(request, headers, hasBody: true);
      if (body != null) {
        request.write(json.encode(body));
      }
      final response = await request.close();
      final responseBody = await response.transform(utf8.decoder).join();
      return ApiResponse.fromJson(
        json.decode(responseBody) as Map<String, dynamic>,
        null,
      );
    } finally {
      client.close();
    }
  }

  void _addHeaders(
    HttpClientRequest request,
    Map<String, String>? extraHeaders, {
    bool hasBody = false,
  }) {
    defaultHeaders.forEach(request.headers.set);
    extraHeaders?.forEach(request.headers.set);
    request.headers.set('Content-Type', 'application/json');
    request.headers.set('Accept', 'application/json');
    request.headers.set('X-Request-Id', _generateRequestId());
  }

  String _generateRequestId() {
    final now = DateTime.now().microsecondsSinceEpoch;
    final random = (now % 100000).toString().padLeft(5, '0');
    return 'BEZA-${now}-$random';
  }
}
